shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -1511,14 +1514,33 @@ void updateSemantics(
       // of the screen in landscape mode. We must handle the translation ourselves for the
       // a11y nodes.
       if (Build.VERSION.SDK_INT >= 23) {
-        WindowInsets insets = rootAccessibilityView.getRootWindowInsets();
-        if (insets != null) {
-          if (!lastLeftFrameInset.equals(insets.getSystemWindowInsetLeft())) {
-            rootObject.globalGeometryDirty = true;
-            rootObject.inverseTransformDirty = true;
+        boolean needsToApplyCutoutInsect = true;
+        if (Build.VERSION.SDK_INT >= 28) {
+          Context context = rootAccessibilityView.getContext();
+          if (context instanceof Activity) {
+            Activity activity = (Activity) context;
+            if (activity.getWindow() != null && activity.getWindow().getAttributes() != null) {
+              switch (activity.getWindow().getAttributes().layoutInDisplayCutoutMode) {
+                case WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES:
+                case WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_ALWAYS:
+                  needsToApplyCutoutInsect = false;
+                case WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER:
+                case WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT:
```

Comment:
it could be simplify.  Maybe, this could work:

```java
if (layoutInDisplayCutoutMode == LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT ||
    layoutInDisplayCutoutMode  == LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER) {
   // code under if (needsToApplyCutoutInsect) {}
}
```

I think the case `activity.getWindow() == null` means that the activity isn't visual, so we don't need to do anything.

The case `activity.getWindow().getAttributes() == null` doesn't appear to be documented in the Android docs.